### PR TITLE
VIH-7691 Command center will display Venues instead of courtroom accounts

### PR DESCRIPTION
**Pull request by ZJianWang**
### JIRA link (if applicable) ###
[VIH-7691](https://tools.hmcts.net/jira/browse/VIH-7691)

### Change description ###
- VHO hearing list selection in command center will display Venues instead of courtroom accounts
- VHO will be able to select multiple venues from the hearing list selection dropdown

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb.UnitTests/Builders/ConferenceResponseBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
+using System;
+using System.Collections.Generic;
+using UserApi.Contract.Responses;
+using VideoApi.Contract.Responses;
+
+namespace VideoWeb.UnitTests.Builders
+{
+    public static class ConferenceResponseBuilder
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb.UnitTests/Builders/ConferenceResponseBuilder.cs:8]: Rename to ConferenceForAdminResponseBuilder

File: VideoWeb/VideoWeb.UnitTests/Controllers/UserDataControllerTest.cs [hunk 2]
```diff
@@ -20,24 +22,25 @@ public class UserDataControllerTest
...
+            var conferences = ConferenceResponseBuilder.BuildData();
 
-            _mocker.Mock<IUserApiClient>().Setup(x => x.GetJudgesAsync()).ReturnsAsync(accounts);
+            _mocker.Mock<IVideoApiClient>().Setup(x => x.GetConferencesTodayForAdminByHearingVenueNameAsync(null, _query.HearingVenueNames)).ReturnsAsync(conferences);
 
-            var result = await _sut.GetCourtRoomsAccounts(usernames);
+            var result = await _sut.GetCourtRoomsAccounts(_query);
 
             var typedResult = (OkObjectResult)result.Result;
             typedResult.Should().NotBeNull();
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb.UnitTests/Controllers/UserDataControllerTest.cs:46]: Can you rename judgeList to something more fitting like courtRoomAccounts and maybe change like "FirstName1" to "Venue1" and "LastName1" to "Room1". Sorry to be picky but it makes the test hard to read.

File: VideoWeb/VideoWeb/Controllers/VenuesController.cs [hunk 3]
```diff
@@ -18,33 +21,35 @@ public class VenuesController : Controller
...
+        ///     Get available courts
         /// </summary>
-        /// <returns>List of judges with hearing scheduled, if any</returns>
-        [HttpGet]
-        [ProducesResponseType(typeof(JudgeNameListResponse), (int)HttpStatusCode.OK)]
-        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
-        [SwaggerOperation(OperationId = "GetDistinctJudgeNames")]
-        public async Task<ActionResult<JudgeNameListResponse>> GetDistinctJudgeNamesAsync()
+        /// <returns>List of courts</returns>
+        [HttpGet("courts", Name = "GetCourts")]
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb/Controllers/VenuesController.cs:38]: Maybe rename GetCourts to GetVenues or vice versa so it is consistent across the swagger operation and logs.

File: VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs [hunk 4]
```diff
@@ -11,23 +11,22 @@ public class CourtRoomsAccountResponseMapperTest : BaseMockerSutTestSetup<CourtR
...
+            var result = _sut.Map(conferences);
 
             result.Should().NotBeNull();
             result.Count.Should().Be(2);
             result[0].CourtRooms.Count.Should().Be(3);
             result[1].CourtRooms.Count.Should().Be(1);
 
-            result[0].Venue.Should().Be("Manual01");
-            result[1].Venue.Should().Be("Manual03");
+            result[0].Venue.Should().Be("FirstName1");
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs:22]: Can you maybe change like "FirstName1" to "Venue1" and "LastName1" to "Room1". Sorry to be picky but it makes the test harder to read.

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.html [hunk 5]
```diff
@@ -4,34 +4,19 @@ <h1 class="govuk-heading-m">{{'venue-list.title' | translate}}</h1>
...
-      placeholder="Choose lists"
-      [(ngModel)]="selectedJudges"
-      [closeOnSelect]="false"
-      (change)="updateSelection()"
-      [clearSearchOnAdd]="true"
-      [selectOnTab]="true"
-    >
+    <ng-select id="venue-allocation-list" class="custom"
+      [attr.aria-label]="'venue-list.allocation-list-label' | translate" [items]="venues" [multiple]="true"
+      bindLabel="name" bindValue="name" placeholder="Choose lists" [(ngModel)]="selectedJudges" [closeOnSelect]="false"
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.html:9]: Can you rename the variable `selectedJudges` in `[(ngModel)]="selectedJudges"` to something more fitting

Comment (robertbennett1998) [reply VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.html:9]: I think this should be something like selectedCourtRooms?

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.spec.ts [hunk 6]
```diff
@@ -21,12 +22,18 @@ describe('VenueListComponent', () => {
...
+    const venueNames: HearingVenueResponse[] = [];
+    const venueName1 = new HearingVenueResponse({ id: 1, name: 'Birmingham' });
+    const venueName2 = new HearingVenueResponse({ id: 2, name: 'Manchester' });
+    const venueName3 = new HearingVenueResponse({ id: 3, name: 'Taylor House' });
+    venueNames.push(venueName1);
+    venueNames.push(venueName2);
+    venueNames.push(venueName3);
+
+    const selectedJudgeNames: string[] = [];
+    selectedJudgeNames.push(venueName1.name);
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.spec.ts:34]: Rename `selectedJudgeNames` to something more fitting

File: VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.ts [hunk 7]
```diff
@@ -35,8 +35,8 @@ export class VenueListComponent implements OnInit {
 
     ngOnInit() {
         this.venueListLoading = false;
-        this.videoWebService.getDistinctJudgeNames().then(response => {
-            this.judges = response.first_names;
+        this.videoWebService.getVenues().subscribe(venues => {
+            this.venues = venues;
             this.selectedJudges = this.judgeAllocationStorage.get();
```
Comment (robertbennett1998) [inline VideoWeb/VideoWeb/ClientApp/src/app/vh-officer/venue-list/venue-list.component.ts:40]: Same as above, rename `selectedJudges` to something more fitting.

File: VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs (hunk 4 above)
Comment (ZJianWang) [reply VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs:22]: I changed the venues and courtnames to firstName and lastNames in

Comment (robertbennett1998) [reply VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs]: The judge name isn't used any more is it? The variable name Venue and CourtRooms was correct. I was talking about the string comparisons. Can you change the variable names back, they aren't descriptive now. Sorry if I wasn't more clear.

Comment (ZJianWang) [reply VideoWeb/VideoWeb.UnitTests/Mappings/CourtRoomsAccountResponseMapperTest.cs]: The Judge name isn't used in admin/venue-list but still used in admin/hearing-list

**Review by robertbennett1998 — DISMISSED** [review]
LGTM. Nice work.

**Review by robertbennett1998 — APPROVED** [review]
LGTM!

**Pull request merged**